core/commands/checkout.py

```diff
@@ -51,21 +46,24 @@ class GsCheckoutNewBranchCommand(WindowCommand, GitCommand):
     Prompt the user for a new branch name, create it, and check it out.
     """
 
-    def run(self):
-        sublime.set_timeout_async(self.run_async)
+    def run(self, base_branch=None):
+        sublime.set_timeout_async(lambda: self.run_async(base_branch))
 
-    def run_async(self, name=""):
-        self.window.show_input_panel(NEW_BRANCH_PROMPT, name, self.on_done, None, None)
+    def run_async(self, base_branch=None):
+        self.base_branch = base_branch
```

Comment:
I am lost. If `base_branch` is None, it is just like previously. Else if `base_branch` is not None, it will checkout a new branch from `base_branch`.